build/assets/scripts/rt-kernel.sh

```diff
@@ -1,55 +1,55 @@
 #!/usr/bin/env bash
 
+######################################################################################
+## NOTE:                                                                            ##
+## THIS IS A TEMPORARY WORKAROUND UNTIL THIS FEATURE IS AVAILABLE VIA MACHINECONFIG ##
+## IT ONLY WORKS WITH RT KERNEL VERSION 4.*                                         ##
+######################################################################################
+
 set -euo pipefail
 
-REPO_DIR="/etc/yum.repos.d"
-RT_REPO="${REPO_DIR}/rt-kernel.repo"
+echo "Mounting OS image"
 
-# Enable yum repo
-if [[ -f $RT_REPO ]]
-then
-    # The env var might have been changed, so always create new rt repo
-    rm $RT_REPO
-fi
+# remove old container
+osContainerName="osContainer"
+set +e
+podman rm -f "$osContainerName" >/dev/null 2>&1
+set -e
 
-mkdir -p $REPO_DIR
-cat > $RT_REPO <<EOF
-[rt]
-baseurl=${RT_REPO_URL}
-gpgcheck=0
-EOF
-
-# update cache
-rpm-ostree refresh-md -f
-
-exit_handler () {
-    exit_code=$?
-    if [[ ${exit_code} -eq 77 ]]; then 
-        echo "No update available, nothing to do"; 
-        exit 0;
-    elif [[ ${exit_code} -eq 100 ]]; then
-        echo "Initiate reboot, touch /var/reboot"
-        touch /var/reboot
-        exit 0;
-    else
-        exit ${exit_code}
-    fi
-}
+# find booted image sha
+sha=$(rpm-ostree status --json | jq -r '.deployments[] | select(.booted == true) | .["custom-origin"][0]' | sed -E "s|.*@(.*)|\1|")
+imageID=$(podman images --digests | grep $sha | awk '{print $4}')
 
-trap exit_handler EXIT
+# create and mount new container
+podman create --net=none --name "$osContainerName" "$imageID" > /dev/null
+kernelDir=$(podman mount "$osContainerName")
```

Comment:
wow, so we're delivering kernel pkgs in containers? interesting.

Reply:
AFAIU this is used to access the packages in the basesystem (so we are not shipping anything anymore). But I may have out of date information.

Reply:
this is complicated, let me try to summarize:
- the installer's bootstrap node installs an "old" os bootimage on cluster nodes, which is just new enough to be able to do everything which is needed on first boot
- then the MCO installs a newer version of the os ("early pivot"). That one is delivered via container image. Maybe you noticed that OCP comes with a `machine-os-content` image: that one contains the os
- with RHCOS version 44 the image contains the RT kernel RPMs. They were just put into the root directory of the image
- so what I do here (stolen from MCO code): mount the image (which is not possible directly, so create (but don't run) a container from it) which contains the os version which is currently booted, and install the RPMs from the mount path.

longer version: https://github.com/openshift/machine-config-operator/blob/master/docs/OSUpgrades.md and others

Reply:
Thanks @slintes great summary. I think it would be great to record this as comment in the `rt-kernel.sh` script itself. WDYT?

Reply:
The operator is for 4.4 only. There we have the RT kernel RPMs already available in the RHCOS image, so no need for a yum repo anymore, but the MCO is not ready to use it yet, see WIP PR [0]. So this is a new temporary workaround for 4.4, unrelated to 4.3.

[0] https://github.com/openshift/machine-config-operator/pull/1330

Reply:
The operator is for 4.4 only. There we have the RT kernel RPMs already available in the RHCOS image, so no need for a yum repo anymore, but the MCO is not ready to use it yet, see WIP PR [0]. So this is a new temporary workaround for 4.4, unrelated to 4.3.

[0] openshift/machine-config-operator#1330

Reply:
@fromanirh ah missed your comment
will do

Reply:
@fromanirh done